spec/core/v2/ics-004-channel-and-packet-semantics/README.md

```diff
@@ -0,0 +1,1059 @@
+
+### Definitions
+
+`get`, `set`, `delete`, and module-system related primitives are as defined in [ICS 24](../ics-024-host-requirements).
+
+A `channel` is a data structure that facilitates exactly-once packet delivery between two blockchains acting as a communication pipeline between specific modules registered on separate chains, allowing for secure, verifiable transmission of packets. Channels can be created and registered to establish a semantic link between two chains and their respective light clients, ensuring that both chains can process and verify the packets exchanged. To establish the root of trust for secure interchain communication with a counterparty chain, each chain MUST register a channel maintaining the necessary counterparty information, such as the channel identifier of the counterparty chain, the light client identifier of the counterparty chain and the path used to store packet flow messages.
+
+```typescript
+interface Channel {
+    clientId: bytes // local light client id of the counterparty chain. 
+    counterpartyChannelId: bytes // counterparty channel id.  
+    keyPrefix: CommitmentPrefix //  key path that the counterparty will use to prove its store packet flow messages.
+}
+```
+
+The `Packet`, `Payload`, `Encoding` and the `Acknowledgement` interfaces are as defined in [packet specification](https://github.com/cosmos/ibc/blob/c7b2e6d5184b5310843719b428923e0c5ee5a026/spec/core/v2/ics-004-packet-semantics/PACKET.md). 
+
+For convenience, following we recall their structures.  
+
+A `Packet`, in the interblockchain communication protocol, is a particular interface defined as follows:
+
+```typescript
+interface Packet {
+    sourceChannelId: bytes, // channel identifier on the source chain. 
+    destChannelId: bytes, // channel identifier on the dest chain.
+    sequence: uint64, // number that corresponds to the order of sent packets.
+    timeout: uint64, // indicates the UNIX timestamp in seconds and is encoded in LittleEndian. It must be passed on the destination chain and once elapsed, will no longer allow the packet processing, and will instead generate a time-out.
+    data: Payload[] // data 
+}
+```
+
+The `Payload` is a particular interface defined as follows:
+
+```typescript
+interface Payload {
+    sourcePort: bytes, // identifies the source application port
+    destPort: bytes, // identifies the dest application port 
+    version: string, // application version
+    encoding: Encoding, // used encoding - allows the specification of custom data encoding among those agreed in the `Encoding` enum
```

Comment:
Application data is just hashed now no? We should drop all references to this `Encoding` interface?